### PR TITLE
Slider component

### PR DESCRIPTION
**Pull request by cafeed28**
Example:
```tsx
<Slider
  markers={[16, 24, 32, 40, 48, 56, 64, 80, 96]}
  stickToMarkers={true}
  defaultValue={48}
  initialValue={32}
  minValue={16}
  maxValue={96}
  onMarkerRender={(e: number) => {
    return `${e}px`;
  }}
/>
```
image

### CODE REVIEW

File: src/renderer/modules/components/Slider.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  initialValue?: number;
+  defaultValue?: number;
+  minValue?: number;
+  maxValue?: number;
+  mini?: boolean;
+  style?: React.CSSProperties;
+  onValueChange?: (e: number) => void;
+  asValueChanges?: (e: number) => void;
+  onValueRender?: (e: number) => string;
+  onMarkerRender?: (e: number) => string;
```
Comment (asportnoy) [inline src/renderer/modules/components/Slider.tsx:16]: Missing the `className` property

File: src/renderer/modules/components/Slider.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { filters, waitForModule } from "../webpack";
+
+interface SliderProps {
+  disabled?: boolean;
+  markers?: number[];
+  stickToMarkers?: boolean;
+  initialValue?: number;
+  defaultValue?: number;
```
Comment (asportnoy) [inline src/renderer/modules/components/Slider.tsx:8]: `defaultValue` doesn't exist

**Review by asportnoy — CHANGES_REQUESTED** [review]
Can you add aliases for the props `initialValue` as `value` and `onValueChange` as `onChange`? (Create a separate component that wraps the one from Discord and apply the values on the original object). That way, it'll work better with the `useSetting` hook.

File: src/renderer/modules/components/Slider.tsx (hunk 2 above)
Comment (cafeed28) [reply src/renderer/modules/components/Slider.tsx:9]: `defaultValue` is green marker

image

File: src/renderer/modules/components/index.ts [hunk 3]
```diff
@@ -91,6 +91,11 @@ export type { TextInputType };
 export let TextInput: TextInputType;
 importTimeout("Input", import("./TextInput"), (mod) => (TextInput = mod.default));
 
+import type { SliderType } from "./Slider";
+export type { SliderType };
+export let Slider: SliderType;
+importTimeout("Input", import("./Slider"), (mod) => (Slider = mod.default));
```
Comment (FedeIlLeone) [inline src/renderer/modules/components/index.ts:97]: Shouldn't "Input" be changed to "Slider"?

File: src/renderer/modules/components/Slider.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+interface SliderProps extends SliderCompProps {
+  value?: number;
+  onChange?: (e: number) => void;
+}
+
+export type SliderType = ReactComponent<SliderProps>;
+
+export default ((props) => {
+  return (
+    <SliderComp {...props} initialValue={props.value} onValueChange={props.onChange}></SliderComp>
```
Comment (asportnoy) [inline src/renderer/modules/components/Slider.tsx:38]: Move the spread to the end so people can still use the original names

**Review by asportnoy — CHANGES_REQUESTED** [review]

File: src/renderer/modules/components/Slider.tsx (hunk 4 above)
Comment (cafeed28) [reply src/renderer/modules/components/Slider.tsx]: Done

File: src/renderer/modules/components/index.ts (hunk 3 above)
Comment (cafeed28) [reply src/renderer/modules/components/index.ts]: Fixed

File: src/renderer/modules/components/Slider.tsx (hunk 1 above)
Comment (cafeed28) [reply src/renderer/modules/components/Slider.tsx:18]: Fixed in [this commit](https://github.com/replugged-org/replugged/pull/366/commits/42ff71b350430157d78a50f5bce59b144e07d1cb#diff-ba27653ecb680e72e999facdb29940cc49686074aee6280329274baf5d104122R14)

**Review by asportnoy — COMMENTED** [review]
Code looks good, will test later today

**Review by asportnoy — APPROVED** [review]

**Pull request merged**